### ISSUE TITLE
About the inconsistency between fake_sample_epoch and gen_image

### ISSUE THREAD

**Issue by taiyangcheer** [3neutronstar/ACGAN-PyTorch#1]
Thank you very much for your modification to the official acgan code, the gennerate_sample.py you added helped me a lot.
But I encountered a problem, part of the image generated by generate_sample.py has some strange colored blocks (just a part), while the sample image generated by main.py does not have this problem.
How can I set it up so that the image generated by generate_sample.py is consistent with the fake_sample_epoch generated by main.py.
looking forward to your reply.

**Comment by 3neutronstar** [comment]
Hello. Thanks for your interest in my code. Could I get your model and example images? (Examples of both with and without colored block one, etc.)

**Comment by taiyangcheer** [comment]
thank you very much for your reply

![5](https://user-images.githubusercontent.com/105720223/198507475-d4de6fc3-3e8f-430e-889d-6de970708a6a.png)（colored block）

![40](https://user-images.githubusercontent.com/105720223/198507674-5a8c2cb5-260b-4a8d-b081-8f16347487ad.png)（without colored block）


![fake_samples_epoch_1999](https://user-images.githubusercontent.com/105720223/198507782-900f0c78-1d51-4cfc-8b4d-4787d3beaa70.png)（generated images by mian.py  ）

![uTools_1666945216467](https://user-images.githubusercontent.com/105720223/198541467-2dae12e6-d096-48cf-9561-d1d3067ad0a9.png) (part of images with colored block)


There are images of the flames of an industrial kiln, they have three combustion states. I use to do classification tasks.
Below are some of my parameter settings：

    parser = argparse.ArgumentParser()
    parser.add_argument('--dataset', default='imagenet', help='cifar10 | imagenet')
    parser.add_argument('--dataroot', default='', help='path to dataset')
    parser.add_argument('--workers', type=int, help='number of data loading workers', default=0)
    parser.add_argument('--batchSize', type=int, default=60, help='input batch size')
    parser.add_argument('--imageSize', type=int, default=128, help='the height / width of the input image to network')
    parser.add_argument('--nz', type=int, default=110, help='size of the latent z vector')
    parser.add_argument('--ngf', type=int, default=128)
    parser.add_argument('--ndf', type=int, default=128)
    parser.add_argument('--niter', type=int, default=200, help='number of epochs to train for')
    parser.add_argument('--lr', type=float, default=0.0002, help='learning rate, default=0.0002')
    parser.add_argument('--beta1', type=float, default=0.5, help='beta1 for adam. default=0.5')
    parser.add_argument('--cuda', default=True, action='store_true', help='enables cuda')
    parser.add_argument('--ngpu', type=int, default=1, help='number of GPUs to use')
    parser.add_argument('--netG', default='', help="path to netG (to continue training)")
    parser.add_argument('--netD', default='', help="path to netD (to continue training)")
    parser.add_argument('--outf', default='', help='folder to output images and model checkpoints')
    parser.add_argument('--manualSeed', type=int,default=1, help='manual seed')
    parser.add_argument('--num_classes', type=int, default=3, help='Number of classes for AC-GAN')
    parser.add_argument('--gpu_id', type=int, default=0, help='The ID of the specified GPU')

transform.normalize(0.5,0.5,0.5)(0.5,0.5,0.5)

Is the appearance of the colored blocks related to the setting of the nz parameter? Can I improve the generated images by deepening the network?

**Comment by 3neutronstar** [comment]
Thank you for the detailed answer. I think the denormalization operation in `utils.py` affects your generated image. In the imagenet, I didn't test `generated_image.py`. If my assumption is correct, I highly recommend you fix the mean and std of the normalization operation into `utils.py` denormalization `mean` and `std` value. I think this affects the image quality a lot.
